### PR TITLE
Devel

### PR DESCRIPTION
**Pull request by SirrahErydya**
Point Cloud Filter can optionally be shifted to the Robot's viewport coordinate system

### CODE REVIEW

File: robosherlock/src/annotation/src/PrimitiveShapeAnnotator.cpp [hunk 1]
```diff
@@ -334,6 +334,14 @@ class PrimitiveShapeAnnotator : public DrawingAnnotator
...
     }
   }
+
+  void calculate_line_inliers(pcl::PointCloud<PointT>::Ptr cloud, pcl::PointIndices::Ptr line_inliers) {
+    // Pick three random points from this cloud and project them to conformal space
+    auto point1 = cloud.get()->points.at(std::rand() % cloud.get()->points.size());
+    auto point2 = cloud.get()->points.at(std::rand() % cloud.get()->points.size());
+    auto point3 = cloud.get()->points.at(std::rand() % cloud.get()->points.size());
+
+  }
```
Comment (bbferka) [inline robosherlock/src/annotation/src/PrimitiveShapeAnnotator.cpp]: where is this used? If not could you remove it please?

File: robosherlock/src/filter/src/PointCloudFilter.cpp [hunk 2]
```diff
@@ -104,7 +134,18 @@ class PointCloudFilter : public DrawingAnnotator
     pass.setFilterFieldName("z");
     pass.filter(*cloud_filtered);
 
-    cas.set(VIEW_CLOUD, *cloud_filtered);
+    pcl::PointCloud<PointT>::Ptr cloud(new pcl::PointCloud<PointT>);
+    if(transform_cloud) {
+        outInfo("Transforming matrix back (this will deeply hurt the CPU...");
```
Comment (bbferka) [inline robosherlock/src/filter/src/PointCloudFilter.cpp]: Alternatively you could check for NaN-s in filtered cloud and set their counterparts in cloud to NaN

File: robosherlock/src/filter/src/PointCloudFilter.cpp [hunk 3]
```diff
@@ -81,14 +87,38 @@ class PointCloudFilter : public DrawingAnnotator
...
+    pcl::PointCloud<PointT>::Ptr cloud_transformed(new pcl::PointCloud<PointT>);
+    tf::Transform viewport;
+
+    if(transform_cloud) {
+      //@Todo This has nothing to do with base frame yet...
+      outInfo("Transforming Point Cloud to Viewport space");
+      tf::StampedTransform camToWorld;
+      tf::StampedTransform worldToCam;
+      camToWorld.setIdentity();
+      if(scene.viewPoint.has()) {
```
Comment (bbferka) [inline robosherlock/src/filter/src/PointCloudFilter.cpp]: you could replace this with a simple tf lookupTransform; [see here example](https://wiki.ros.org/tf/Tutorials/Writing%20a%20tf%20listener%20%28C%2B%2B%29) and use  a param for setting the frame;

**Comment by bbferka** [comment]
@SirrahErydya please change the name of the PR so it reflects the content.

**Comment by bbferka** [comment]
@SirrahErydya new features should go on separate branches; The plane multiple plane detection should not show up in this PR;

**Comment by SirrahErydya** [comment]
I created a new pull request from a branch specific to this feature. I still need to fix some things on the new branch (it is explained in the description of the request), but I already sorted out unused code lines and features that should belong to another branch.

So this pull request is deprecated and can be ignored or deleted.

**Pull request closed without merging**